File: Scoring/Motor de Scoring V1.py
def calcular_scoring(datos_cliente: dict) -> dict:
  
    puntaje = 50
    motivo = []

    ingreso = datos_cliente["ingreso"]
    deuda = datos_cliente["deuda"]
    edad = datos_cliente["edad"]
    historial = datos_cliente["historial"].lower()
    empleo_anios = datos_cliente["empleo_anios"]
    monto = datos_cliente["monto"]

    #relación deuda e ingreso
    ratio_dti = deuda / ingreso if ingreso > 0 else 1
    if ratio_dti < 0.3:
        puntaje += 20
    elif ratio_dti <= 0.5:
        puntaje += 10
    else:
        puntaje -= 20
        motivo.append("Relación deuda/ingreso demasiado alta")

    # rango de edad
    if edad < 21:
        puntaje -= 30
        motivo.append("Edad menor a 21 años")
    elif 21 <= edad <= 25:
        puntaje += 5
    elif 26 <= edad <= 60:
        puntaje += 15
    else:
        puntaje -= 10
        motivo.append("Edad fuera del rango ideal")

    # analizar historial crediticio
    if historial == "excelente":
        puntaje += 30
    elif historial == "bueno":
        puntaje += 20
    elif historial == "regular":
        puntaje += 5
    elif historial == "malo":
        puntaje -= 25
        motivo.append("Historial crediticio deficiente")

    # analisis de estabilidad laboral
    if empleo_anios < 1:
        puntaje -= 10
        motivo.append("Poca estabilidad laboral")
    elif 1 <= empleo_anios <= 3:
        puntaje += 5
    else:
        puntaje += 15

    # monto/ingreso
    ratio_monto = monto / ingreso if ingreso > 0 else 10
    if ratio_monto < 3:
        puntaje += 15
    elif ratio_monto <= 6:
        puntaje += 5
    else:
        puntaje -= 20
        motivo.append("Monto solicitado demasiado alto para sus ingresos")

    if puntaje >= 70:
        decision = "Aprobado"
    elif 60 <= puntaje < 70:
        decision = "Revisión manual"
        motivo.append("Caso requiere análisis adicional")
    else:
        decision = "Rechazado"

    return {
        "puntaje": puntaje,
        "decision": decision,
        "motivo": motivo if motivo else ["Cumple criterios"]
    }


if __name__ == "__main__":

File: Scoring/Motor de Scoring V1.js
function calcularPuntaje({ edad, ingresos, historial_crediticio, monto_solicitado, plazo }) {
  let puntaje = 0;
  let motivos = [];

  if (edad >= 21 && edad <= 65) {
    puntaje += 30;
  } else {
    puntaje -= 20;
    motivos.push("Edad fuera de rango (21-65)");
  }

  if (ingresos >= 500000) {
    puntaje += 40;
  } else if (ingresos >= 300000) {
    puntaje += 20;
  } else {
    puntaje -= 20;
    motivos.push("Ingresos insuficientes");
  }

  switch (historial_crediticio) {
    case "bueno":
      puntaje += 30;
      break;
    case "regular":
      puntaje += 10;
      break;
    case "malo":
      puntaje -= 30;
      motivos.push("Historial crediticio negativo");
      break;
    default:
      motivos.push("Historial no informado");
  }

  if (monto_solicitado > ingresos * 0.5) {
    puntaje -= 20;
    motivos.push("Monto solicitado alto respecto a ingresos");
  }

  if (plazo > 60) {
    puntaje -= 10;
    motivos.push("Plazo excesivo");
  }

  const decision = puntaje >= 50 ? "Aprobado" : "Rechazado";

  if (decision === "Aprobado" && motivos.length === 0) {
    motivos.push("Cumple criterios");
  }

  return { puntaje, decision, motivos };
}

module.exports = { calcularPuntaje };
